toolkit/pallets/block-production-log/src/test.rs

```diff
@@ -0,0 +1,138 @@
+use super::*;
+use frame_support::{assert_ok, traits::UnfilteredDispatchable};
+use mock::*;
+use sp_consensus_slots::Slot;
+use sp_runtime::BoundedVec;
+
+fn make_id(i: u64) -> [u8; 32] {
+	let mut id = [0u8; 32];
+	id[0..8].copy_from_slice(&i.to_le_bytes());
+	id
+}
+
+#[test]
+fn first_append_should_succeed() {
+	new_test_ext().execute_with(|| {
+		let call = Call::<Test>::append { block_producer_id: make_id(1) };
+		assert_ok!(call.dispatch_bypass_filter(RuntimeOrigin::none()));
+
+		assert_eq!(Log::<Test>::get().to_vec(), vec![(Slot::from(1001000), make_id(1))]);
+	})
+}
+
+#[test]
+fn append_to_end_of_log() {
+	new_test_ext().execute_with(|| {
+		let existing_item = vec![(Slot::from(100), make_id(1))];
+		LatestSlot::<Test>::put(Slot::from(100));
+		Log::<Test>::put(
+			BoundedVec::<_, <mock::Test as pallet::Config>::MaxLogSize>::truncate_from(
+				existing_item,
+			),
+		);
+
+		let call = Call::<Test>::append { block_producer_id: make_id(2) };
+		assert_ok!(call.dispatch_bypass_filter(RuntimeOrigin::none()));
+
+		assert_eq!(
+			Log::<Test>::get().to_vec(),
+			vec![(Slot::from(100), make_id(1)), (Slot::from(1001000), make_id(2))]
+		);
+	})
+}
+
+#[test]
+fn can_not_append_same_item_twice() {
+	new_test_ext().execute_with(|| {
+		let existing_item = vec![(Slot::from(100), make_id(1))];
+		LatestSlot::<Test>::put(Slot::from(100));
+		Log::<Test>::put(
+			BoundedVec::<_, <mock::Test as pallet::Config>::MaxLogSize>::truncate_from(
+				existing_item,
+			),
+		);
```

Comment:
for BoundedVec setup consider BoundedSlice
```suggestion
		let binding = vec![(Slot::from(100), make_id(1))];
		let existing_item = BoundedSlice::try_from(binding.as_slice()).unwrap();
		LatestSlot::<Test>::put(Slot::from(100));
		Log::<Test>::put(existing_item);
```

Reply:
Done, thanks